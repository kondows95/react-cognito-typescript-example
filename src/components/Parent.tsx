import React, {PropsWithChildren} from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../store';

const Parent: React.FC<PropsWithChildren<{}>> = (props) => {
  return (
    <Provider store={store}>
        <Router>
          {props.children}
        </Router>
    </Provider>
  );
};
export default Parent;