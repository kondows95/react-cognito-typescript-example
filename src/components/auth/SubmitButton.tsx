import React, {PropsWithChildren} from 'react';
import { Button, CircularProgress}  from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  progress: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    marginTop: -12,
    marginLeft: -12,
  },
  item: {
    [theme.breakpoints.down("xs")]: { 
      padding: theme.spacing(1, 0, 1), //TXL
    }, 
    [theme.breakpoints.up("sm")]: { 
      padding: theme.spacing(2, 0, 1, 1), //TRBL
    },
  },
}));

type Props = { disabled: boolean }

const SubmitButton: React.FC<PropsWithChildren<Props>> = (props) => {
  const classes = useStyles();

  let loading = null;
  if (props.disabled) {
    loading = <CircularProgress size={20} color="primary" className={classes.progress} />;
  }
  return (
    <Button
        fullWidth 
        {...props} 
        variant="contained" 
        color="primary"
        type="submit"
      > 
        {props.children || "Submit"}
        {loading}
      </Button>
  )
};
export default SubmitButton;