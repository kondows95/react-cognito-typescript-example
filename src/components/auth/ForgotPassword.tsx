import React, {MouseEvent, FormEvent, ChangeEvent} from 'react';
import { Box, Container, Link, TextField}  from '@material-ui/core';
import FormTitle from './FormTitle';
import SubmitButton from './SubmitButton';

type Props = {
  authState: string
  changeAuthState: (state: string) => void
  forgotPassword: (email: string) => void
  loading: boolean
  error: string
};

const ForgotPassword: React.FC<Props> = ({ authState, changeAuthState, loading, error, forgotPassword }) => {
  type Form = {　[key: string]: string　};
  const [form, setForm] = React.useState<Form>({email:"", password:""});
  
  type FormKey = "email" | "password";
  const handleChangeValue = (fieldName: FormKey) => (event: ChangeEvent<HTMLInputElement>) => {
    const newForm = {...form};
    newForm[fieldName] =  event.currentTarget.value;
    setForm(newForm);
  };

  const handleSubmit = (event: FormEvent)  => {
    event.preventDefault();
    forgotPassword(form['email']);
  };


  const handleSignIn = (event: MouseEvent)  => {
    event.preventDefault();
    changeAuthState('signIn');
  };

  const content = (
    <Container component="main" maxWidth="xs">
      <form onSubmit={handleSubmit}>
        <Box display="flex" flexDirection="column" mt={8}>
          <FormTitle>Reset Your Password</FormTitle>
          <Box display="flex" justifyContent="center" fontWeight={600} color="error.main">
            {error}
          </Box>
          <Box width="100%" my={2}>
            <TextField
              autoComplete="email"
              type="email"
              onChange={handleChangeValue("email")}
              value={form.email}
              label="Email Address"
              variant="outlined"
              required
              fullWidth
              inputProps={{"data-testid":"email"}}
            />
          </Box>

          <Box width="100%" mt={4} mb={2} className="relative">
            <SubmitButton　data-testid="submitButton" disabled={loading}>
              Send Code
            </SubmitButton>
          </Box>
          <Box width="100%" my={2}>
            <Link data-testid="signInLink" href="#" variant="body2" onClick={handleSignIn}>
              Back to Sign in
            </Link>
          </Box>
        </Box>
      </form>
    </Container>
  );

  return (authState === 'forgotPassword') ? content : null;
};

export default ForgotPassword;