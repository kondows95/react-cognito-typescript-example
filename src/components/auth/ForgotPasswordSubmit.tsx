import React, {FormEvent, ChangeEvent} from 'react';
import { Box, Container, TextField}  from '@material-ui/core';
import FormTitle from './FormTitle';
import SubmitButton from './SubmitButton';

type Props = {
  authState: string | null;
  forgotPasswordSubmit: (email: string, confirmationCode:string, password:string) => void;
  loading: boolean;
  error: string;
  email: string;
};

const ForgotPasswordSubmit: React.FC<Props> = ({ authState, loading, error, email, forgotPasswordSubmit }) => {
  type Form = {　[key: string]: string　};
  const [form, setForm] = React.useState<Form>({email:"", password:"",confirmationCode:""});
  
  type FormKey = "email" | "password" | "confirmationCode";
  const handleChangeValue = (fieldName: FormKey) => (event: ChangeEvent<HTMLInputElement>) => {
    const newForm = {...form};
    newForm[fieldName] =  event.target.value;
    setForm(newForm);
  };

  const handleSubmit = (event: FormEvent)  => {
    event.preventDefault();
    forgotPasswordSubmit(email,form['confirmationCode'],form['password']);
  };

  const content = (
    <Container component="main" maxWidth="xs">
      <form onSubmit={handleSubmit}>
        <Box display="flex" flexDirection="column" mt={8}>
          <FormTitle>Please confirm your email</FormTitle>
          <Box display="flex" justifyContent="center" fontWeight={600} color="error.main">
            {error}
          </Box>
          <Box width="100%" my={2}>
            <TextField
              label="Confirmation Code"
              onChange={handleChangeValue("confirmationCode")} 
              value={form.confirmationCode}
              variant="outlined"
              required
              fullWidth
              inputProps={{"data-testid":"confirmationCode"}}
            />
          </Box>
          
          <Box width="100%" my={2}>
            <TextField
              label="New Password"
              type="password"
              autoComplete="new-password"
              onChange={handleChangeValue("password")}
              value={form.password}
              variant="outlined"
              required
              fullWidth
              inputProps={{"data-testid":"password"}}
            />
          </Box>
          
          <Box width="100%" mt={4} mb={2} className="relative">
            <SubmitButton　data-testid="submitButton" disabled={loading}>
              Confirm
            </SubmitButton>
          </Box>
        </Box>
      </form>
    </Container>
  );

  return (authState === 'forgotPasswordReset') ? content : null;
};

export default ForgotPasswordSubmit;