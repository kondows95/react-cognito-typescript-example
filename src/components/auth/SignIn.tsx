import React, {MouseEvent, FormEvent, ChangeEvent} from 'react';
import { Box, Container, Link, TextField}  from '@material-ui/core';
import FormTitle from './FormTitle';
import SubmitButton from './SubmitButton';

type Props = {
  authState: string
  changeAuthState: (state: string) => void
  signIn: (email: string, password: string) => void
  loading: boolean
  error: string
};

const SignIn: React.FC<Props> = ({ authState, changeAuthState, signIn, loading, error }) => {
  type Form = {　[key: string]: string　};
  const [form, setForm] = React.useState<Form>({email:"", password:""});
  
  type FormKey = "email" | "password";
  const handleChangeValue = (fieldName: FormKey) => (event: ChangeEvent<HTMLInputElement>) => {
    const newForm = {...form};
    newForm[fieldName] =  event.currentTarget.value;
    setForm(newForm);
  };
  
  const handleSubmit = (event: FormEvent)  => {
    event.preventDefault();
    signIn(form['email'], form['password']);
  };

  const handleForgotPassword = (event: MouseEvent)  => {
    event.preventDefault();
    changeAuthState('forgotPassword');
  };

  const content = (
    <Container component="main" maxWidth="xs">
      <form onSubmit={handleSubmit}>
        <Box display="flex" flexDirection="column" mt={8}>
          <FormTitle>Sign In</FormTitle>
          <Box display="flex" justifyContent="center" fontWeight={600} color="error.main">
            {error}
          </Box>
          <Box width="100%" my={2}>
            <TextField
              autoComplete="email"
              type="email"
              onChange={handleChangeValue('email')}
              value={form.email}
              label={loading? " " : "Email Address"}
              variant="outlined"
              required
              fullWidth
              inputProps={{"data-testid":"email"}}
            />
          </Box>
          <Box width="100%" my={2}>
            <TextField
              label={loading? " " : "Password"}
              type="password"
              autoComplete="current-password"
              onChange={handleChangeValue('password')} 
              value={form.password}
              variant="outlined"
              required
              fullWidth
              inputProps={{"data-testid":"password"}}
            />
          </Box>
          <Box width="100%" mt={4} mb={2} className="relative">
            <SubmitButton　data-testid="submitButton" disabled={loading}>
              Sign In
            </SubmitButton>
          </Box>
          <Box width="100%" my={2}>
            <Link data-testid="forgotPasswordLink" href="#" variant="body2" onClick={handleForgotPassword}>
              Forgot password?
            </Link>
          </Box>
        </Box>
      </form>
    </Container>
  );
  return (authState === 'signIn') ? content : null;
};

export default SignIn;