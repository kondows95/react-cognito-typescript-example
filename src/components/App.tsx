import React from 'react';
import { MuiThemeProvider } from '@material-ui/core/styles';
import { BrowserRouter } from 'react-router-dom';
import { CssBaseline } from '@material-ui/core';
import theme from '../theme';
import Header from '../containers/Header';
import Login from '../containers/Login';
import { CognitoUser } from '@aws-amplify/auth';

type Props = {
  user: CognitoUser | null
  fetchAuthedUser: () => void
  refreshToken: () => void
};

const App: React.FC<Props> = ({ user, fetchAuthedUser, refreshToken }) => {
  const isFirstRef = React.useRef(true);
  React.useEffect(() => {
    if (isFirstRef.current) {
      isFirstRef.current = false;
      fetchAuthedUser();
    }
  });

  /*React.useEffect(() => {
    const timer = window.setInterval(() => {
      refreshToken();
    }, 1); 
    return () => { 
      window.clearInterval(timer);
    };
  });*/

  const contents = <Header />;

  return (
    <MuiThemeProvider theme={theme}>
      <BrowserRouter>
        <CssBaseline />
        {user ? contents : <Login />}
      </BrowserRouter>
    </MuiThemeProvider>
  );
};

export default App;

