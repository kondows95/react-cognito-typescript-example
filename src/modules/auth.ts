//import { Dispatch, Reducer } from 'redux';
import { Auth } from 'aws-amplify';
import { AppState, AuthState } from './index';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from "redux";
import { CognitoUser } from '@aws-amplify/auth';

const initialState: AuthState = {
  authState: 'signIn',
  user: null,
  email: "",
  error: "",
  loading: false,
};

//=============================================================================
//Reducers
//=============================================================================
export default (state: AuthState = initialState, action: AnyAction): AuthState => {
  switch (action.type) {
    case 'AUTH_SYSTEM_ERROR':
      return {
        ..._getCommonState(state),
        error: action.payload,
        loading: false
      };
    case 'AUTH_BEGIN_LOADING':
      return {
        ..._getCommonState(state),
        loading: true
      };
    case 'AUTH_INIT':
      return {
        ...initialState
      };
    case 'AUTH_CHANGE_AUTH_STATE':
      return {
        ..._getCommonState(state),
        authState: action.payload
      };
    case 'AUTH_FETCH_AUTHED_USER':
      return {
        ..._getCommonState(state),
        user: action.payload
      };
    case 'AUTH_SIGN_IN_SUCCESS':
      return {
        ..._getCommonState(state),
        user: action.payload,
        authState: "",
      };
    case 'AUTH_FORGOT_PASSWORD_SUCCESS':
      return {
        ..._getCommonState(state),
        authState: 'forgotPasswordReset',
        email: action.payload,
      };
    default:
      return state;
  }
};

const _getCommonState = (state: AuthState) => ({
  ...state, 
  error: "",
  loading: false,
});

//=============================================================================
//Actions
//=============================================================================
export type Actions = ReturnType<
  typeof fetchAuthedUserSuccess 
  | typeof authInit 
  | typeof authBeginLoading
  | typeof authError
  | typeof changeAuthState
  | typeof authSignInSuccess
  | typeof authForgotPasswordSuccess
>

export const fetchAuthedUserSuccess = (user: CognitoUser) => ({
  type: 'AUTH_FETCH_AUTHED_USER',
  payload: user
});

export const authInit = () => ({
  type: 'AUTH_INIT',
});

export const authBeginLoading = () => ({
  type: 'AUTH_BEGIN_LOADING',
});

export const authError = (err: any) => ({
  type: 'AUTH_SYSTEM_ERROR',
  payload: err.message || err
});

export const changeAuthState = (value: string) =>  ({
  type: 'AUTH_CHANGE_AUTH_STATE',
  payload: value
});

export const authSignInSuccess = (user: CognitoUser) =>  ({
  type: 'AUTH_SIGN_IN_SUCCESS',
  payload: user
});

export const authForgotPasswordSuccess = (email: string) =>  ({
  type: 'AUTH_FORGOT_PASSWORD_SUCCESS',
  payload: email
});

//=============================================================================
//Async Operations
//=============================================================================
type Dispatch = ThunkDispatch<AppState, undefined, AnyAction>;

export const refreshToken = () => {
  return async (dispatch: Dispatch, getState: () => AppState) => {
    try {
      const cognitoUser = await Auth.currentAuthenticatedUser();
      const currentSession = await Auth.currentSession();
      cognitoUser.refreshSession(currentSession.getRefreshToken(), (err: any, session: any) => {
        //const { idToken, refreshToken, accessToken } = session;
        // do whatever you want to do now :)
      });
    }
    catch(err) {
      dispatch(authError(err));
    }
  };
};

export const fetchAuthedUser = () => {
  return async (dispatch: Dispatch, getState: () => AppState) => {
    dispatch(authBeginLoading());
    try {
      const user = await Auth.currentAuthenticatedUser();
      dispatch(fetchAuthedUserSuccess(user));
    }
    catch {
      dispatch(authInit());
    }
  };
};

export const signOut = () => {
  return async (dispatch: Dispatch, getState: () => AppState) => {
    dispatch(authInit());
    try {
      await Auth.signOut();
    }
    catch(err) {
      //No error message is set here.
      //If you set error message here, the message is always displayed.
      dispatch(authError(err));
    }
  };
};

export const signIn = (email: string, password: string) => {
  return async (dispatch: Dispatch, getState: () => AppState) => {
    dispatch(authBeginLoading());
    try {
      const user = await Auth.signIn(email, password);
      dispatch(authSignInSuccess(user));
    }
    catch(err) {
      dispatch(authError(err));
    }
  };
};

export const forgotPassword = (email: string) => {
  return async (dispatch: Dispatch, getState: () => AppState) => {
    dispatch(authBeginLoading());
    try {
      await Auth.forgotPassword(email);
      dispatch(authForgotPasswordSuccess(email));
    }
    catch(err) {
      dispatch(authError(err));
    }
  };
};

export const forgotPasswordSubmit = (email: string, code: string, password: string) => {
  return async (dispatch: Dispatch, getState: () => AppState) => {
    dispatch(authBeginLoading());
    try {
      await Auth.forgotPasswordSubmit(email, code, password);
      dispatch(authInit());
    }
    catch(err) {
      dispatch(authError(err));
    }
  };
};