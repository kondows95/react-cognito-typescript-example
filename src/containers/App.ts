import { connect } from 'react-redux';
import App from '../components/App';
import { AppState, AppDispatch } from '../types';
import { fetchAuthedUser,refreshToken } from '../modules/auth';

export const mapStateToProps = (state: AppState) => ({
  user: state.auth.user,
  authState: state.auth.authState,
  loading: state.auth.loading,
});

export const mapDispatchToProps = (dispatch: AppDispatch) => ({
  fetchAuthedUser: () => dispatch(fetchAuthedUser()),
  refreshToken: () => dispatch(refreshToken())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);