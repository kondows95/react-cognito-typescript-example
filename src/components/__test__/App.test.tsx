/*global jest,expect*/
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import Parent from '../Parent';
import App from '../App';
import TestUtil from '../../TestUtil';
import { CognitoUser } from '@aws-amplify/auth';

let refreshToken: jest.Mock;
let fetchAuthedUser: jest.Mock;
beforeEach(() => {
  refreshToken = jest.fn();
  fetchAuthedUser = jest.fn();
});

describe("User is null",()=>{
  let util: TestUtil;
  beforeEach(() => {
    util = new TestUtil(
      <Parent> 
        <App
          user={null}
          fetchAuthedUser={fetchAuthedUser}
          refreshToken={refreshToken}
        />
      </Parent>
    );
  });

  it('Snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });

  it('Header is not displayed', () => {
    expect(util.render.container).not.toHaveTextContent("Management Console");
  });

  it('Re-rendering', () => {
    for (let i=0; i<2; i++) {
      util.render.rerender(
        <Parent> 
          <App
            user={{} as CognitoUser}
            fetchAuthedUser={fetchAuthedUser}
            refreshToken={refreshToken}
          />
        </Parent>
      );
      expect(fetchAuthedUser).toHaveBeenCalledTimes(1);
    }
  });
});

describe("User is not null",()=>{
  const user = {} as CognitoUser;
  let util: TestUtil;
  beforeEach(() => {
    util = new TestUtil(
      <Parent> 
        <App
          user={user}
          fetchAuthedUser={fetchAuthedUser}
          refreshToken={refreshToken}
        />
      </Parent>
    );
  });

  it('Snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });

  it('Header is displayed', () => {
    expect(util.render.container).toHaveTextContent("Management Console");
  });
});
