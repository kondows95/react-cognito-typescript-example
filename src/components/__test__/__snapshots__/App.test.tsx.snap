// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User is not null Snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-684 MuiAppBar-colorPrimary mui-fixed"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="Open drawer"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-685 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root makeStyles-title-686 MuiTypography-h6 MuiTypography-noWrap"
      >
        Management Console
      </h6>
      <div
        class="MuiBox-root MuiBox-root-779"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <button
            aria-controls="long-menu"
            aria-haspopup="true"
            aria-label="More"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            data-testid="openMenuButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`User is null Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <main
      class="MuiContainer-root MuiContainer-maxWidthXs"
    >
      <form>
        <div
          class="MuiBox-root MuiBox-root-10"
        >
          <div
            class="MuiBox-root MuiBox-root-11"
          >
            <div
              class="MuiBox-root MuiBox-root-12"
            >
              <div
                class="MuiBox-root MuiBox-root-13"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="none"
                  >
                    <path
                      d="M0 0h24v24H0V0z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      opacity=".87"
                    />
                  </g>
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiBox-root MuiBox-root-23"
            >
              Sign In
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-24"
          />
          <div
            class="MuiBox-root MuiBox-root-25"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
              >
                Email Address
                <span
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="email"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="email"
                  required=""
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-88 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-89"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-90"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
              >
                Password
                <span
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="current-password"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="password"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-88 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-89"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-91 relative"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
              data-testid="submitButton"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Sign In
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiBox-root MuiBox-root-125"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
              data-testid="forgotPasswordLink"
              href="#"
            >
              Forgot password?
            </a>
          </div>
        </div>
      </form>
    </main>
  </div>
</DocumentFragment>
`;
